### PR TITLE
Support multiple readers in the mapping-template component

### PR DESCRIPTION
**Pull request by AuPath**
Adds an endpoint to the mapping template component _"mapt://readers?..."_ that expects as input an exchange whose body contains a `Map<String,Reader>` of `Readers`.

If accepted, it should be merged only after https://github.com/cefriel/mapping-template/pull/27.

### CODE REVIEW

File: camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java [hunk 1]
```diff
@@ -148,36 +144,46 @@ else if ((params.resourceCustomFunctions() == null) && (params.customFunctions !
                 formatter = Util.createFormatter(params.formatterFormat());
             }
 
+            TemplateExecutor templateExecutor;
```
Comment (marioscrock) [inline camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java:147]: Move TemplateExecutor as a field of the class MaptTemplateProducer to reuse the same instance across multiple executions of the same route

**Review by marioscrock — CHANGES_REQUESTED** [review]

File: camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java [hunk 2]
```diff
@@ -93,92 +93,140 @@ private static boolean validateParams(OperationParams params) {
...
+
     }
-    private static void execute(OperationParams params, Exchange exchange, String inputFormat) throws Exception {
 
-        if (validateParams(params)) {
-            TemplateExecutor templateExecutor = new TemplateExecutor();
-	    
-	    Reader reader = getReaderFromExchange(exchange, inputFormat, params.formatterFormat(), params.verboseReader());
+    private static TemplateExecutor templateExecutor(OperationParams params, Exchange exchange, String inputFormat) throws Exception {
+        // which custom template functions to use ???
```
Comment (marioscrock) [inline camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java:104]: Move as an issue in the repo explaining current behaviour and changes to be discussed

File: camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java [hunk 3]
```diff
@@ -93,92 +93,140 @@ private static boolean validateParams(OperationParams params) {
...
+                                ResourceAccessor.open(params.query(), exchange),
+                                outputFilePath);
+                    }
                     exchange.getMessage().setBody(resultFilesPaths, List.class);
+
                 } else {
-                    Path resultFilePath =
-                            templateExecutor.executeMapping(reader, ResourceAccessor.open(params.template(), exchange), outputFilePath, templateMap, formatter, usedTemplateFunctions);
+                    Path resultFilePath;
+                    if (reader != null) {
```
[the remark this answers is not in the document]
Comment (AuPath) [reply camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java:188]: Changed so as to reflect changes to the mapping-template in https://github.com/cefriel/mapping-template/pull/30

File: camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java [hunk 4]
```diff
@@ -93,92 +93,140 @@ private static boolean validateParams(OperationParams params) {
...
+
                     exchange.getMessage().setBody(resultFilePath, String.class);
                 }
             }
             // if filename is not specified then the result of applying template is stored in the exchange body as string
             else {
                 if (params.query() != null) {
-                    Map<String,String> result = templateExecutor.executeMappingParametric(reader, ResourceAccessor.open(params.template(), exchange), ResourceAccessor.open(params.query(), exchange), templateMap, formatter, usedTemplateFunctions);
+                    Map<String,String> result;
+                    if (reader != null) {
```
Comment (marioscrock) [inline camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java:202]: Here and in other invocations to the `mapping-template`, avoid repeated code blocks. Change as:

```java
Map<String, Reader> readers; //Check where defined and if we need to define it again
if (reader != null)
    readers = Map.of("reader", reader);
result = templateExecutor.executeMappingParametric(readers,
                                ResourceAccessor.open(params.template(), exchange),
                                ResourceAccessor.open(params.query(), exchange));
```

File: camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java [hunk 5]
```diff
@@ -93,92 +93,140 @@ private static boolean validateParams(OperationParams params) {
...
+                        result = templateExecutor.executeMappingParametric(readers,
+                                ResourceAccessor.open(params.template(), exchange),
+                                ResourceAccessor.open(params.query(), exchange));
+                    }
                     exchange.getMessage().setBody(result, Map.class);
 
                 } else {
-                    String result = templateExecutor.executeMapping(reader, ResourceAccessor.open(params.template(), exchange), templateMap, formatter, usedTemplateFunctions);
+                    String result;
+                    if (reader != null) {
```
Comment (marioscrock) [inline camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java:216]: Ideally, define `readers` map at the beginning and then perform calls without checking `reader ` every time

**Review by marioscrock — CHANGES_REQUESTED** [review]

**Review by hamzabhatti5253 — APPROVED** [review]

**Pull request merged**